src/@dvcorg/gatsby-theme-iterative/components/Page/getLayoutComponent.ts

```diff
@@ -0,0 +1,26 @@
+import MainLayout from '@dvcorg/gatsby-theme-iterative/src/components/MainLayout'
+import DocumentationLayout from '@dvcorg/gatsby-theme-iterative/src/components/DocumentationLayout'
+import AlertLandingLayout from '@dvcorg/gatsby-theme-iterative/src/components/AlertLandingLayout'
+import BlogLayout from '../../../../components/Blog/Layout'
+
+const getLayoutComponent = (props: {
+  pageContext: {
+    is404: boolean
+    isBlog: boolean
+    isDocs: boolean
+    isAlertLanding: boolean
+  }
+}) => {
+  if (!props.pageContext.is404) {
+    if (props.pageContext.isBlog) {
+      return BlogLayout
+    } else if (props.pageContext.isDocs) {
+      return DocumentationLayout
+    } else if (props.pageContext.isAlertLanding) {
+      return AlertLandingLayout
+    }
+  }
+  return MainLayout
+}
+
+export default getLayoutComponent
```

Comment:
This broken out layout logic file lets us reuse our weird layout system with whatever components or logic we want with minimal need for shadowing- the only reason we need it here is the blog layout, sites without a blog likely won't need to shadow this at all!

That said, this kind of system that imports all the components into one super-component is generally considered a bad idea for code splitting reasons. Obviously we did it for a reason, but I think looking into a better solution is warranted in the future.